lib/vtls/mbedtls.c

```diff
@@ -46,6 +46,8 @@
 #include <mbedtls/ctr_drbg.h>
 #include <mbedtls/sha256.h>
 
+#include <mbedtls/debug.h>
```

Comment:
Shouldn't you add a `#ifdef MBEDTLS_DEBUG`?

Reply:
From the code it looks like the idea is that you manually "#define" instead of the "#undef" in https://github.com/curl/curl/blob/924ae8653d49b387491d75b3db498a4d378503e9/lib/vtls/mbedtls.c#L119
Assuming that is correct your suggestion would not fix the issue.

I didn't know if it was permissible to add the include file in the block that follows line 119?
Or let mbedtls/config.h decide if MBEDTLS_DEBUG is set or not. I.e. deleting line 119?

Reply:
I think the `#undef` of the debug define probably should be removed. Or at least moved to the top of the file so that it is easier to find.

Reply:
I think the proper way to do it would be to include mbedtls/config.h . But I am not 100% certain that it does not introduce some regression. So I have moved the ```#undef``` to the top.

Reply:
I think undefing stuff we haven't defined is a bad idea and will create lots of confusion. If people invoke the compilation with extra CPP definitions, we shouldn't ignore them IMO.

Reply:
But again, I am not able to judge on this as I am not into the library 😊

Reply:
What do you think about the latest commit? It combines the previous functionality with your suggestion.

Reply:
Do you mean d06f3e19a323bcd82fafa2da41fdcf587b93069c?
If so it's not perfect. Double slash comments are not allowed on the one hand, on the second hand I don't think we should leave unused Code commented out, especially when it has never been in use